lib/commands/general.js

```diff
@@ -68,14 +68,22 @@ commands.isKeyboardShown = async function () {
 
 commands.hideKeyboard = async function () {
   let {isKeyboardShown, canCloseKeyboard} = await this.adb.isSoftKeyboardPresent();
-  if (!isKeyboardShown) {
-    throw new Error("Soft keyboard not present, cannot hide keyboard");
-  }
-
-  if (canCloseKeyboard) {
-    return await this.back();
+  if (isKeyboardShown) {
+    if (canCloseKeyboard) {
+      // Press escape
+      await this.adb.keyevent(111);
+    }
+    try {
```

Comment:
I don't understand: if we can't close the keyboard, we still wait and wait until it is just closed on its own?

Reply:
yes, it is just in case we are in the middle of some running animation